amethyst_rendy/src/camera.rs

```diff
@@ -524,19 +699,29 @@ impl Camera {
         ))
     }
 
+    /// Creates a `Projection::CustomMatrix` with the matrix provided.
+    pub fn custom_matrix(
+        matrix: Matrix4<f32>,
+        clip_fn: &'static (dyn Fn(&CustomMatrix) -> (f32, f32) + Send + Sync),
```

Comment:
This looks really bad. I'd say that custom matrix projection should simply not have `near` and `far` as part of the API. Other option is just to use the formulas from perspective projection. Notice that in arbitrary matrices the clip planes doesn't have to be perpendicular. You can in fact force the near plane to be in any orientation you want, so just a float doesn't really cut it.